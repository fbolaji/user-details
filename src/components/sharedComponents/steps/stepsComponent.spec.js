import React from 'react';
import { render, create, cleanup } from '@testing-library/react';
import StepsComponent from './stepsComponent';

const mockStepsList=[
	{stepIndex: 1, cssClass: 'inactive', page: 'user'},
	{stepIndex: 2, cssClass: 'inactive', page: 'privacy'},
	{stepIndex: 3, cssClass: 'inactive', page: 'done'}
];
const renderedHTML = `<div data-testid="test-steps" class="steps-container"><ul class="steps"><li class="is-active completed">user</li><li class="inactive">privacy</li><li class="inactive">done</li></ul></div>`;

describe('StepComponent', () => {
	afterEach(cleanup);

	it('Should render and match snapshot', () => {
		const { container } = render(<StepsComponent stepsList={mockStepsList} />);

		expect(container).toMatchSnapshot();
	});

	it('Should render list of steps', () => {
		const { container, getByText } = render(<StepsComponent stepsList={mockStepsList} />);

		expect(getByText(/user/)).toBeInTheDocument();
		expect(getByText(/privacy/)).toBeInTheDocument();
		expect(getByText(/done/)).toBeInTheDocument();

		expect(container.innerHTML).toMatch(renderedHTML);
	});
});