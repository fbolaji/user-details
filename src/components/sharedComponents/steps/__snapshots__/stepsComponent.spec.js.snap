// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepComponent Should render and match snapshot 1`] = `
<div>
  <div
    class="steps-container"
    data-testid="test-steps"
  >
    <ul
      class="steps"
    >
      <li
        class="is-active completed"
      >
        user
      </li>
      <li
        class="inactive"
      >
        privacy
      </li>
      <li
        class="inactive"
      >
        done
      </li>
    </ul>
  </div>
</div>
`;

exports[`StepComponet Should render and match snapshot 1`] = `
<div>
  <div
    class="steps-container"
    data-testid="test-steps"
  >
    <ul
      class="steps"
    >
      <li
        class="is-active"
      >
        user
      </li>
      <li
        class="inactive"
      >
        privacy
      </li>
      <li
        class="inactive"
      >
        done
      </li>
    </ul>
  </div>
</div>
`;

exports[`StepComponet Should render list of steps 1`] = `
<div>
  <div
    class="steps-container"
    data-testid="test-steps"
  >
    <ul
      class="steps"
    >
      <li
        class="is-active"
      >
        user
      </li>
      <li
        class="inactive"
      >
        privacy
      </li>
      <li
        class="inactive"
      >
        done
      </li>
    </ul>
  </div>
</div>
`;

exports[`StepComponet Should render steps list 1`] = `
<div>
  <div
    class="steps-container"
    data-testid="test-steps"
  >
    <ul
      class="steps"
    >
      <li
        class="is-active"
      >
        user
      </li>
      <li
        class="inactive"
      >
        privacy
      </li>
      <li
        class="inactive"
      >
        done
      </li>
    </ul>
  </div>
</div>
`;
